### PR TITLE
ci: tweak ci runs to decrease wait time of devs

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by asikowitz — APPROVED** [review]
Batches look good

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -27,8 +27,8 @@ jobs:
         command:
           [
             # metadata-ingestion and airflow-plugin each have dedicated build jobs
-            "./gradlew build -x :metadata-ingestion:build -x :metadata-ingestion:check -x docs-website:build -x :metadata-integration:java:spark-lineage:test -x :metadata-io:test -x :metadata-ingestion-modules:airflow-plugin:build -x :metadata-ingestion-modules:airflow-plugin:check -x :datahub-frontend:build -x :datahub-web-react:build --parallel",
-            "./gradlew :datahub-frontend:build :datahub-web-react:build --parallel",
+            "metadata_ingestion",
```
Comment (hsheth2) [inline .github/workflows/build-and-test.yml:30]: The name is inaccurate - this does everything except metadata ingestion, airflow, docs site, and frontend

File: metadata-ingestion/tests/integration/snowflake/test_snowflake.py [hunk 2]
```diff
@@ -30,6 +30,8 @@
 from tests.integration.snowflake.common import FROZEN_TIME, default_query_results
 from tests.test_helpers import mce_helpers
 
+pytestmark = pytest.mark.integration_batch_2
```
Comment (hsheth2) [inline metadata-ingestion/tests/integration/snowflake/test_snowflake.py:33]: I think batch 2 is already the slowest - let’s use batch 1 for some of these instead

Comment (anshbansal) [reply metadata-ingestion/tests/integration/snowflake/test_snowflake.py:33]: https://github.com/datahub-project/datahub/actions/runs/6405004395 It looks like the fastest as per the latest run

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (anshbansal) [reply .github/workflows/build-and-test.yml:30]: good catch. Fixing

**Pull request merged**